### PR TITLE
fix: do not clone request for actions

### PR DESCRIPTION
**Pull request by jacob-ebey**
This is to avoid a stream halting bug in node-fetch

### CODE REVIEW

**Review by mcansh — APPROVED** [review]

**Comment by jraoult** [comment]
Would probably fix https://github.com/remix-run/remix/issues/972 ?

**Review by kentcdodds — APPROVED** [review]
This makes sense to me, but I'd prefer @mjackson to merge since he may know more of the implications here.

**Comment by mjackson** [comment]
This makes sense to me too, since we the main reason to clone a request is so that the body can be used in multiple handlers. But we only ever call a single handler on action requests, so we should be ok to not clone it here.

I think @jraoult is right, this should fix #972 as well. But I'd love to see some tests around handling large file uploads to make sure that this indeed fixes the problem.

**Comment by kentcdodds** [comment]
Agreed! A test would be great 👍

File: integration/file-uploads-test.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        "app/routes/file-upload.jsx": js`
+          import { Form, unstable_parseMultipartFormData as parseMultipartFormData, useActionData } from "remix";
+          import { uploadHandler } from "~/fileUploadHandler";
+
+          export let action = async ({ request }) => {
+            try {
+              let formData = await parseMultipartFormData(request, uploadHandler);
+
+              let file = formData.get("file");
+              console.log({ file });
```
Comment (kentcdodds) [inline integration/file-uploads-test.ts:37]:
```suggestion
```

**Review by kentcdodds — APPROVED** [review]
Killer 👍

**Comment by arnaudambro** [comment]
related to #1164

**Pull request merged**